### PR TITLE
API routes For PetType data and Pet data for each PetType

### PR DESCRIPTION
**Pull request by benjamin-z-l**
This branch contains the database seeder and schema for Pets and PetTypes, the corresponding models and API endpoints.

### CODE REVIEW

File: server/db/Seeder.js [hunk 1]
```diff
@@ -1,14 +1,90 @@
 import pg from "pg"
-import path from "path"
-import LineReader from "line-reader"
-import { fileURLToPath } from "url"
+// import path from "path"
+// import LineReader from "line-reader"
+// import { fileURLToPath } from "url"
```
Comment (dovakeen118) [inline server/db/Seeder.js]: these comments could be removed

File: server/models/Pet.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    //       const petsData = result.rows
+    //       const pets = petsData.map(pet => new this(pet))
+
+    //       return pets
+
+    //     } catch (error) {
+    //         console.log(error)
+    //         throw (error)
+    //     }
+    // }
```
Comment (dovakeen118) [inline server/models/Pet.js]: Not sure if these methods were meant to be commented out? Typically we don't want to merge incomplete/ unused code

File: server/models/PetType.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    //       const relatedPetsData = result.rows
+    //       const relatedPets = relatedPetsData.map(pet => new Pet(pet))
+
+    //       return relatedPets
+
+    //     } catch (error) {
+    //         console.log(error)
+    //         throw (error)
+    //     }
+    // }
```
Comment (dovakeen118) [inline server/models/PetType.js]: We can either comment these methods in if they are to be used, otherwise we should remove them and add them in when they are needed for a feature - looks like your Router is only using the first 2 methods, in which case these unused methods should be removed for this current feature!

**Review by dovakeen118 — COMMENTED** [review]
Some unused code to be removed for this feature!

File: server/models/PetType.js (hunk 3 above)
Comment (benjamin-z-l) [reply server/models/PetType.js]: OK, all commented out code has been deleted!

File: server/models/PetType.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    static async findByType(type) {
+        const petFile = await import('./Pet.js')
+        const Pet = petFile.default
+        let petTypeId
+        if (type === "cats") {
+            petTypeId = 1
+        } if (type === "dogs") {
+            petTypeId = 2
```
Comment (bkincart) [inline server/models/PetType.js]: We should never hardcode ids into our code -- we don't know for sure if our database will always look like that!

Instead, update this code to query the database by type and find the record itself in your database, then read its id.

We're also running into an issue where conventionally, based on name, I'd expect this method to return a `PetType` object, but instead, it's returning a list of `Pet`s.

OPTION 1:
1. Update this method to query the `pet_types` table by type and find the record itself in your database, then read its id.
2. Rename the method `findPetsByType` to be clear that it's returning `Pet` objects.

OPTION 2: Even better: we could change this method to be more conventional by doing the following:
1. `findByType` should return a `PetType` object itself. Query the db by `type` column and return the PetType, much like `findById` would work
2. Then, create a `.pets()` association method which is an _instance_ method rather than a static method, and returns the `Pet`s for a given `PetType`

**Review by bkincart — CHANGES_REQUESTED** [review]
One more method that needs updating, and then this will be good to go!

**Comment by benjamin-z-l** [comment]
Made some changes to findPetsByType, not a perfect solution, but it's no longer hard coded and will allow us to proceed.

File: server/db/Seeder.js [hunk 5]
```diff
@@ -1,14 +1,87 @@
...
+         await pool.query(queryString, [pet.name, pet.img_url, pet.age, pet.vaccination_status, pet.adoption_story, pet.available_for_adoption, pet.petType.id])
+       }
+       console.log("Seeding is complete")
+       pool.end()
+
+     } catch (error) {
+       console.log(error)
+       pool.end()
+     }
   }
```
Comment (safang99) [inline server/db/Seeder.js:85]: We can first focus on just the `pet_types`. any other tables should not be included in this PR

File: server/db/schema.sql [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+CREATE TABLE adoptable_pets (
+    id SERIAL PRIMARY KEY,
+    name VARCHAR(255) NOT NULL,
+    img_url VARCHAR(255) NOT NULL,
+    age INTEGER,
+    vaccination_status BOOLEAN NOT NULL,
+    adoption_story TEXT NOT NULL,
+    available_for_adoption BOOLEAN NOT NULL,
+    pet_type_id INTEGER REFERENCES pet_types
+);
```
Comment (safang99) [inline server/db/schema.sql:20]: We can first focus on just the `pet_types`. any other tables should not be included in this PR.  We can just remove the second CREATE TABLE command

File: server/models/Pet.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+        this.imgUrl = imgUrl || img_url
+        this.age = age
+        this.vaccinationStatus = vaccinationStatus || vaccination_status
+        this.adoptionStory = adoptionStory || adoption_story
+        this.availableForAdoption = availableForAdoption || available_for_adoption
+        this.petTypeId = petTypeId || pet_type_id
+    }
+}
+
+export default Pet
```
Comment (safang99) [inline server/models/Pet.js:17]: We won't need this for `pet_types` either

File: server/models/PetType.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+      const query = "SELECT * FROM adoptable_pets JOIN pet_types ON pet_type_id = pet_types.id WHERE type = $1;"
+      const result = await pool.query(query, [type])
+      const petsData = result.rows
+      const pets = petsData.map(pet => new Pet(pet))
+      return pets
+    } catch (error) {
+      console.log(error)
+      throw (error)
+    }
+  }
```
Comment (safang99) [inline server/models/PetType.js:40]: We can remove 27-39 as it is out of scope

File: server/routes/api/v1/petTypesRouter.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+petTypesRouter.get("/:type", async (req, res) => {
+  try {
+    const type = req.params.petType
+    const pets = await PetType.findPetsByType(type)
+    res.status(200).json({ pets: pets })
+  } catch (error) {
+    console.log(error)
+    res.status(500).json({ error: error })
+  }
+})
```
Comment (safang99) [inline server/routes/api/v1/petTypesRouter.js:26]: We only need the first endpoint for the first feature so we can remove 17-25

File: server/routes/clientRouter.js [hunk 10]
```diff
@@ -2,9 +2,10 @@ import express from "express"
 
 const router = new express.Router()
 
-const clientRoutes = []
+const clientRoutes = []
```
Comment (safang99) [inline server/routes/clientRouter.js:5]: Make sure to update this with only the path needed for the first feature.  It's important to work slowly and only add what we need when we need it

**Review by safang99 — CHANGES_REQUESTED** [review]
The content looks good but we will need to remove the stuff that is not related to the first feature concerning `pet_types`.  We will also need to include the corresponding React code for this feature
